File: CSC13122-LT-UNG-DUNG-QUAN-LY-2/DataBase/models/laptopspec.js
'use strict';
const {
  Model
} = require('sequelize');
module.exports = (sequelize, DataTypes) => {
  class LaptopSpec extends Model {
    /**
     * Helper method for defining associations.
     * This method is not a part of Sequelize lifecycle.
     * The `models/index` file will call this method automatically.
     */
    static associate(models) {
      // define association here
    }
  }
  LaptopSpec.init({
    SKU: {
      type: DataTypes.STRING,
      allowNull: false,
      unique: true
    },
    VariantID: DataTypes.STRING,
    LaptopID: DataTypes.STRING,
    CPU: DataTypes.STRING,
    GPU: DataTypes.STRING,
    RAM: DataTypes.INTEGER,
    Storage: DataTypes.INTEGER,
    StorageType: DataTypes.STRING,
    Color: DataTypes.STRING,
    ImportPrice: {
      type: DataTypes.DECIMAL,
      allowNull: false
    },
    Price: DataTypes.DECIMAL,
    StockQuantity: DataTypes.INTEGER
  }, {
    sequelize,
    modelName: 'LaptopSpec',
  });
  return LaptopSpec;
};

File: CSC13122-LT-UNG-DUNG-QUAN-LY-2/StormPC/StormPC.Core/Models/Products/LaptopSpec.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using StormPC.Core.Models.Orders;

namespace StormPC.Core.Models.Products;

[Table("LaptopSpecs")]
public class LaptopSpec
{
    [Key]
    public string VariantID { get; set; } = null!;

    [Required]
    [MaxLength(50)]
    public string SKU { get; set; } = null!;

    public string LaptopID { get; set; } = null!;

    [Required]
    [MaxLength(100)]
    public string CPU { get; set; } = null!;

    [MaxLength(100)]
    public string? GPU { get; set; }

    [Required]
    public int RAM { get; set; }

    [Required]
    public int Storage { get; set; }

    [Required]
    [MaxLength(20)]
    public string StorageType { get; set; } = null!;

    [MaxLength(50)]
    public string? Color { get; set; }

    [Required]
    [Column(TypeName = "decimal(18,2)")]
    public decimal ImportPrice { get; set; }

    [Required]
    [Column(TypeName = "decimal(18,2)")]
    public decimal Price { get; set; }

    [Required]
    public int StockQuantity { get; set; }

    // Thuộc tính navigation
    public Laptop Laptop { get; set; } = null!;
    public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
} 